### DISCUSSION TITLE
Ender 3 Pro 4.2.7

### DISCUSSION THREAD

**Discussion by seder001** [mriscoc/Ender3V2S1#521] [Ideas]
Hello everyone,

did someone try this firmware with an ender 3 pro with 4.2.7 board?
Maybe I need to upgrade the display first, but just asking.

**Comment by mriscoc** [comment]
https://github.com/mriscoc/Ender3V2S1/wiki#requirements

**Reply by seder001** [reply]
How am I able to tell if a screen is DWIN or DACAI?
I was going to buy this below to upgrade my ender 3 pro, will that work as well?
https://de.aliexpress.com/item/1005002344513406.html?spm=a2g0o.cart.0.0.46d34ae4CZhV9Q&mp=1&gatewayAdapt=glo2deu

**Reply by mriscoc** [reply]
You must open the screen unit and compare with https://github.com/mriscoc/Ender3V2S1/wiki/How-to-update-the-display

**Reply by seder001** [reply]
I waited for my BLV Upgrade and the screen, after doing this complete faceover I tried this version on my Ender 3 Pro 4.2.7 with BL touch: Ender3V2-427-BLT-20221002 and got it running, but nothing worked.

every homing killed the printer :D

**Reply by mriscoc** [reply]
Your 4.2.7 board could have different stepper motor drivers. So, a custom compile could be necessary.

**Reply by seder001** [reply]
Actually it was my fault I forgot, that I had a 4.2.2 Version. After switching to the right Version, the Motors moved, but the endstops are not being recognized.

Therefore I kill the printer with every homeing.

**Reply by seder001** [reply]
I will try the custom compile as you mentioned

**Reply by mriscoc** [reply]
> Actually it was my fault I forgot, that I had a 4.2.2 Version. After switching to the right Version, the Motors moved, but the endstops are not being recognized.
> 
> Therefore I kill the printer with every homeing.

The firmware has a utility to test the end-stops in the Advanced settings menu, if your motors move the firmware version could be correct.
